### ISSUE TITLE
[BUG] Custom `reporter_mask` being ignored by the R package

### ISSUE THREAD

**Issue by jonjoncardoso** [latentnetworks/vimure#92]
A bug reported by @danielRedhead. 

The R package seems to be ignoring a custom `reporter_mask`:

```r
# Define dimensions
L <- 1  # Number of slices
N <- 275   # Dimension size

unique_nodes <- dplyr::union(edgelist$ego, edgelist$alter) %>% unique()

ego <- unique_nodes
alter <- unique_nodes
reporter <- unique_nodes
reporters <- unique(edgelist$reporter)  # Assuming edgelist$reporter contains the list of reporters

# Create an empty array filled with zeros
reporter_mask <- array(0, dim = c(1, N, N, N),
                       dimnames = list("layer", ego, alter, reporter))

# Populate the array according to the specified conditions
for (l in 1:L) {
  for (i in 1:N) {
    for (j in 1:N) {
      for (m in 1:length(reporters)) {
        if (ego[i] == reporters[m] | alter[j] == reporters[m] & ego[i] != alter[j]) {
          reporter_mask[l, i, j, m] <- 1
        }
      }
    }
  }
}
```

**Comment by jonjoncardoso** [comment]
Work on this is ongoing, @danielRedhead.

I created a small reproducible case with a synthetic network based on the code you provided (in https://github.com/latentnetworks/vimure/commit/0fc073b228e76fb1852615ae6165fde825850fa0) and it took me a while to figure out what was going on. 

[This test passes](https://github.com/latentnetworks/vimure/commit/0fc073b228e76fb1852615ae6165fde825850fa0#diff-f83537a7a3874c18af02284d59fe9171d62bd5d1d56dc1559d75f24d3caf56b6R174), indicating that reporters' mask is correctly passed to the Python code when I customise it so that only egos can report on their outgoing ties. The issue here seems to be related to the fact that Python is 0-indexed whereas R is 1-indexed. I had always assumed that reticulate handled this transparently for us, but it doesn't seem to be the case. 

I'll come back to it later.

**Comment by jonjoncardoso** [comment]
The issue arises from the ordering of nodes in the edgelist.

When setting the reporters' mask by index, as demonstrated below, it assumes that `i`, `j`, and `m` follow the same order as our `unique_nodes` and `unique_reporters` vectors, which isn't always guaranteed.

```r
reporter_mask[l, i, j, m] <- 1
```

To address this, @eapower, @danielRedhead, and I have decided to introduce a different parameter called **reporting_scenario** to our `vimure()` function. This parameter will accept one of the following values:

1. Reporter reports only ties they are involved in (as ego)
2. Reporter reports only ties they are involved in (as ego and as alter) -- the double-sampling case that motivated this work
3. Report on any ties (even those they are not involved in)
4. Custom scenario (TBD)

Our code, inside the vimure function, would then control the ordering of nodes.

**IMPORTANT:** We will then replace the `R` parameter, which currently contains the reporter mask matrix, with a `reporters` list.

**Comment by jonjoncardoso** [comment]
Closed by #93
